docs/zkapps/o1js/bitwise-operations.mdx

```diff
@@ -0,0 +1,303 @@
+---
+title: Bitwise Gadgets
+hide_title: true
+description: Bitwise operations in o1js are implemented as gadgets.
+keywords:
+  - gadgets
+  - bitwise operations
+  - o1js
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Bitwise Operations
+
+In o1js, bitwise operations are implemented as [gadgets](/zkapps/o1js/gadgets).
+
+- [and()](#and)
+- [not()](#not)
+- [xor()](#xor)
+- [leftShift()](#leftshift)
+- [rightShift()](#rightshift)
+- [rotate()](#rotate)
+- [rangeCheck64()](#rangecheck64)
+- [multiRangeCheck()](#multirangecheck)
+- [compactMultiRangeCheck()](#compactmultirangecheck)
+
+See the type declaration for [Gadgets](/zkapps/o1js-reference/modules#gadgets) in the o1js Reference documentation.
+
+## and()
+
+```ts
+and(a: Field, b: Field, length: number) => Field
+```
+
+The bitwise `and()` gadget on Field elements is equivalent to the [bitwise AND (&)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_AND) operator in JavaScript. 
+
+The `and()` gadget:
```

Comment:
@garwalsh  and I talked about how we can "write" gadget names and landed on following how we write about methods, using tick marks and parentheses. Does this Markdown syntax work for the docs?

whatever we do, let's do it consistently throughout

Reply:
The proposed convention is:
- When referring to the gadgets themselves we use "and()" to reflect how they appear in code
- When referring to a boolean operation we use AND